.buildkite/auditbeat/auditbeat-pipeline.yml

```diff
@@ -128,49 +131,60 @@ steps:
   - group: "Auditbeat Extended Tests"
     key: "auditbeat-extended-tests"
     if: build.env("BUILDKITE_PULL_REQUEST") == "false" ||  build.env("GITHUB_PR_LABELS") =~ /.*(macOS|arm|integrations).*/
-
     steps:
-      - label: ":linux: Auditbeat Ubuntu Integration Tests"
+      - label: ":linux: Auditbeat Linux x86_64 Integration Tests -- {{matrix.image}}"
         key: "auditbeat-extended-integ-tests"
         if: build.env("GITHUB_PR_LABELS") =~ /.*integrations.*/
         command: |
           set -euo pipefail
+          echo "using kernel $(uname -r)"
           cd auditbeat
           mage build integTest
         retry:
           automatic:
             - limit: 3
         agents:
           provider: "gcp"
-          image: "${IMAGE_UBUNTU_X86_64}"
+          image: "{{matrix.image}}"
           machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
         artifact_paths:
           - "auditbeat/build/*.xml"
           - "auditbeat/build/*.json"
         notify:
           - github_commit_status:
-              context: "auditbeat: Linux Integration Tests"
+              context: "auditbeat: Linux x86_64 Integration Tests"
```

Comment:
By using a matrix and having a static context we'll be facing the issue that all failures will be collapsed under the same GH status notification.

For example, right now, with passing tests, the notification looks like:

![image](https://github.com/elastic/beats/assets/1754575/d40f4ea9-a1b5-4387-92f6-9cb7b29a496f)

If we click the link it takes us to:

![image](https://github.com/elastic/beats/assets/1754575/5be97e48-a102-4e77-890b-58e805a3dafd)

which represents both `Auditbeat Linux x86_64 Integration Tests -- family/platform-ingest-beats-ubuntu-2004` and `Auditbeat Linux x86_64 Integration Tests -- family/platform-ingest-beats-ubuntu-2204`

So if there is a failure, clicking on the link will take the user to the top of the matrix section, not the specific failed job. Looking at the [official BK docs](https://buildkite.com/docs/pipelines/build-matrix) it may not be possible to add `{{matrix.image}}` -- or some additional property lime `{{matrix.name}}` -- under `notify: ["github_commit_stats": {"context": "the message ... {{matrix.property}}"}]` (maybe worth testing to be 100% sure).

In any case, my main comment here is to raise awareness about the behavior and that with failures it won't be able to quickly see in the status section which specific image failed.